sim/hunter/item_sets_pve.go

```diff
@@ -191,3 +191,170 @@ var ItemSetGiantstalkerPursuit = core.NewItemSet(core.ItemSet{
+				Duration: time.Second * 5,
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					for _, spell := range hunter.MeleeSpells {
+						if spell != nil {
+							spell.DamageMultiplier *= 1.20
+						}
+					}
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					for _, spell := range hunter.MeleeSpells {
+						if spell != nil {
+							spell.DamageMultiplier /= 1.20
+						}
+					}
+				},
+				OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+					if !spell.ProcMask.Matches(core.ProcMaskMeleeMHSpecial) {
+						return
+					}
+
+					aura.Deactivate(sim)
+					if spell == hunter.RaptorStrike {
+						aura.Activate(sim)
+					}
+				},
+			})
+
+			core.MakePermanent(hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Melee 2P Bonus Trigger",
+				OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+					if spell == hunter.RaptorStrike {
+						procAura.Activate(sim)
+					}
+				},
+			}))
+		},
+		// Increases main hand weapon damage by 5%.
+		4: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			hunter.PseudoStats.MHDpsMultiplier += 0.05
+		},
+		// Your periodic damage has a 5% chance to reset the cooldown on one of your Strike abilities. The Strike with the longest remaining cooldown is always chosen.
+		6: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			core.MakePermanent(hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Melee 6P Bonus Trigger",
+				ActionID: core.ActionID{SpellID: 467334},
+				OnPeriodicDamageDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if sim.Proc(0.05, "T2 Melee 6PC Strike Reset") {
+						maxSpell := hunter.RaptorStrike
+
+						for _, strike := range hunter.Strikes {
+							if strike.TimeToReady(sim) > maxSpell.TimeToReady(sim) {
+								maxSpell = strike
+							}
+						}
+
+						maxSpell.CD.Reset()
+						aura.Activate(sim) // used for metrics
+					}
+				},
+			}))
+		},
+	},
+})
+
+var ItemSetDragonstalkerPursuit = core.NewItemSet(core.ItemSet{
+	Name: "Dragonstalker's Pursuit",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your Aimed Shot deals 20% more damage to targets afflicted by one of your trap effects.
+		2: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+
+			procAura := hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Ranged 2P Bonus",
+				ActionID: core.ActionID{SpellID: 467235},
+				Duration: time.Second * 1, // Duration doesn't really matter as the duration is set to the duration of the trap debuff that applies this aura
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					hunter.AimedShot.DamageMultiplier *= 1.20
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					hunter.AimedShot.DamageMultiplier /= 1.20
+				},
+			})
```

Comment:
This won't quite work because it's a general 20% damage without actually checking if the target is afflicted by one of the traps. This is mostly important for immolation trap since it's a single-target trap. I think this probably has to be done inside of the aimed shot code itself unfortunately